pallets/vesting-reward/src/lib.rs

```diff
@@ -0,0 +1,295 @@
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+pub mod weight;
+
+use frame_support::{decl_module, decl_storage, decl_event, decl_error,
+    Parameter,
+    ensure
+};
+use frame_system::{ensure_root, RawOrigin};
+use sp_runtime::traits::{Member, MaybeSerialize, Convert, CheckedDiv};
+use frame_support::sp_runtime::traits::{AtLeast32BitUnsigned, MaybeSerializeDeserialize, Saturating, CheckedSub};
+use frame_support::dispatch::Codec;
+use frame_support::dispatch::fmt::Debug;
+use frame_support::traits::{Get, Currency, ExistenceRequirement::{KeepAlive}, EnsureOrigin};
+use frame_support::dispatch::DispatchResult;
+use crowdloan_claim::reward::Reward;
+use sp_runtime::{ModuleId, traits::{AccountIdConversion, StaticLookup}, Perbill};
+pub use weight::WeightInfo;
+use sp_runtime::traits::Zero;
+
+pub trait Config: frame_system::Config + pallet_vesting::Config {
+    /// This module emits events, and hence, depends on the runtime's definition of event
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+
+    /// The balance type of the relay chain
+    type RelayChainBalance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
+        MaybeSerializeDeserialize + Debug +
+        Into<BalanceOf<Self>>;
+
+    /// AccountId of the relay chain
+    type RelayChainAccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + MaybeSerialize + Ord
+        + Default;
+
+    /// Admin or the module. I.e. this is necessary in cases, where the vesting parameters need
+    /// to be changed without an additional initialization.
+    type AdminOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Weight information for extrinsics in this pallet
+    type WeightInfo: WeightInfo;
+}
+
+type BalanceOf<T> = <<T as pallet_vesting::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+const MODULE_ID: ModuleId = ModuleId(*b"rwd/vest");
+
+decl_event!{
+    pub enum Event<T> where
+        AccountId = <T as frame_system::Config>::AccountId,
+        DirectReward = BalanceOf<T>,
+        VestedReward = BalanceOf<T>,
+        BlockNumber = <T as frame_system::Config>::BlockNumber,
+    {
+        /// Event emitted when a reward claim was processed successfully.
+        RewardClaimed(AccountId, DirectReward, VestedReward),
+        /// Event triggered when the reward module is ready to reward contributors
+        /// [vesting_start, vesting_period, conversion_rate, direct_payout_ratio]
+        RewardModuleInitialized(BlockNumber, BlockNumber, Perbill, Perbill),
+        /// Direct payout ratio for contributors has been updated
+        UpdateDirectPayoutRatio(Perbill),
+        /// Conversion rate from relay to native token has been updated
+        UpdateConversionRate(Perbill),
+        /// Vesting period has been updated
+        UpdateVestingPeriod(BlockNumber),
+        /// Start of vesting has been updated
+        UpadteVestingStart(BlockNumber),
```

Comment:
```suggestion
        UpdateVestingStart(BlockNumber),
```

Reply:
Noted. Will be done by the end of the week.